### DISCUSSION TITLE
Welcome to mpi-sppy Discussions!

### DISCUSSION THREAD

**Discussion by bknueven** [Pyomo/mpi-sppy#228] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jasperstoter** [comment]
I have just started using mpi-sppy and  I have build an  two stage Extensive Form stochastic model. I was wondering if there is a way to acces the solutions that the model creates. In the farmer example ef.get_root_solution() and ef.get_objective_value() to get the root decision variables and objective. However i would also like to see what the second stage decision variables are. Are there any examples this being done?

**Reply by jasperstoter** [reply]
I manage to acces the scenario solutions for the extensive form model by using the following command 
`[*output] = ef.scenarios()`
which gives a list of tuples with the name and model for every scenario

**Comment by DLWoodruff** [comment]
Take a look at this and let me know if this does the trick:
https://mpi-sppy.readthedocs.io/en/latest/access_solutions.html

**Reply by jasperstoter** [reply]
Thank you for the fast reply!  For the Extensive form the method described worked and i could acces the results. However I did not understand how to use the wheelspinner and write_tree_solution functions that you suggested. Now i am trying to solve the model using progressive hedging for which i cant use the EF method. Could you give a example on how to do this?

**Reply by DLWoodruff** [reply]
Most of the examples have a file called E_cylinders.py that relies on E.py. For example,
examples/sizes/sizes_cylinders.py and examples/sizes/sizes.py
The documentation on readthedocs also has some background information.

**Comment by jasperstoter** [comment]
Is it possible to add a time limit to the progressive hedging algorithm? For pyomo models the time limit can be passed to the server in the solver options. I however cant find a similar function of the progressive hedging algorithm

**Comment by DLWoodruff** [comment]
You can give an iteration limit, but no one has yet implemented a time
limit.
  Dave
